plc4j/api/src/main/java/org/apache/plc4x/java/api/messages/specific/TypeSafePlcReadResponse.java

```diff
@@ -84,7 +116,9 @@ public TypeSafePlcReadResponse(TypeSafePlcReadRequest<T> request, List<ReadRespo
         if (type == null) {
             type = Object.class;
         }
-        return new TypeSafePlcReadResponse(new TypeSafePlcReadRequest(type, plcReadResponse.getRequest()), responseItems);
+        // Warning: no validation that everything in the response is a T.
+        // Verified that everything in the response was the same type, but why bother?
```

Comment:
one of the contracts of TypeSafePlcReadResponse is that all values types need to be of the same type. `org/apache/plc4x/java/api/messages/specific/TypeSafePlcReadResponse.java:40`.
The for loop above is not a check, it's more a find my missing parameter `Class class` ;)